src/iter/mod.rs

```diff
@@ -809,6 +811,14 @@ pub trait IndexedParallelIterator: ParallelIterator {
         zip::new(self, zip_op.into_par_iter())
     }
 
+    /// Interleave elements of this iterator and the other given iterator.
```

Comment:
Your PR is more descriptive than these docs -- please elaborate!  An example would be nice too.